client/src/components/common/boost/BoostCalibration.tsx

```diff
@@ -0,0 +1,86 @@
+import React , {useCallback, useState} from 'react';
+import {Button, Card, InputGroup, FormControl} from 'react-bootstrap';
+import {setCalibration as sendCalibrationValue, resetCalibration} from 'api/common/powerModel';
+// import {useSensorData} from 'api/v2/sensors/data';
+
+/**
+ * Boost distance claibration input form
+ * 
+ * @returns {React.Component} Component
+ */
+function BoostCalibrationInput() {
+    const [calibrationValue, setCalib] = useState('');
+
+    const handleCalibrationChange = useCallback(
+        (event) => {
+        setCalib(event.target.value);
+        },
+        [setCalib],
+    );
+
+    const onSet = useCallback(
+        () => {
+          const calibValue: number = parseInt(calibrationValue, 10);
+          if (Number.isNaN(calibValue)) {
+            alert("Please enter an integer value only!");
+          }
+          else {
+            console.log(calibValue);
+            sendCalibrationValue(parseInt(calibrationValue, 10));
```

Comment:
Rather than calling the API, for now instead call some callback that is passed in as a prop. This way it's very easy to test if the code is working from within storybooks.

Reply:
Actually this is something I was confused about. I kind of felt just publishing to dashboard was quite easy to test the code rather than using storybook for which you have to write more code and use callbacks. I think I remember you guys discussing why we went with storybook in one of our weekly meetings, but I can't recall. Was there any specific benefits?

Reply:
Good question. Writing our components like this allows us to completely detach the UI from the logic, meaning that both can be tested independently. This means we can test the components we write without needing to write any logic beforehand. In this case, some of the power model stuff happens to already be written, but were that not the case, testing whether the buttons work would be more difficult. In components with complex logic driving them, it also makes it much easier to test difficult to access states, or to be able to quickly check that your component looks right for a bunch of different inputs if you make a small change.

Although we haven't currently written any front-end unit tests, having the callbacks as props also makes it possible to automatically test the functionality of the component. For example, we could have a unit test that enters "100" as the calibration distance, clicks the "set" button, and then asserts that it got the correct data submitted in the callback function. This wouldn't be possible if the component just sent the data straight back to the server (which wouldn't even be running for a unit test)

Reply:
Yeah true that makes sense. With the unit testing though we can easily mock the server side and do the checking that way, but I do get your point!

Reply:
So is the **calibration difference** published by `BOOST` or the actual **calibrated distance**. Your first comment here  made me think that `BOOST` only provides the calibration difference and so atm the function takes this as input and adds the given instance to display the calibrated distance. Is this the right way around?

Reply:
Sorry for the confusion, I just checked and right now it publishes the calibration difference. I'm going to be modifying that section of boost shortly though (right now it publishes a query string, going to change it to json) and I could change it if needed.